src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp

```diff
@@ -106,6 +106,11 @@ int XMC_Flasher::xclUpgradeFirmware(std::istream& tiTxtStream) {
     int retries = 5;
     int ret = 0;
 
+    if (!hasSC()) {
+        std::cout << "ERROR: SC is not present on platform" << std::endl;
+        return -EINVAL;
+    }
+
     if (!isXMCReady())
```

Comment:
Have you tested on U50, U200?

The logic should be if no XMC, SC won't be there, so we should check XMC first.

You changed the logic to if has SC then check XMC, we might miss some XMC status.

Reply:
We have  ""if (!hasXMC()) return false;"" check in hasSC() function. So, we are checking XMC first and then checking sc_presence state.

Reply:
ok